src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveWorkflowOperations.cs

```diff
@@ -189,12 +189,10 @@ internal class RInteractiveWorkflowOperations : IRInteractiveWorkflowOperations
 
         public void TryRunShinyApp () {
             _shinyRunningTask = Task.Run(async () => {
-                try {
-                    using (var e = await _workflow.RSession.BeginInteractionAsync()) {
-                        await e.RespondAsync("library(shiny)" + Environment.NewLine + "runApp()" + Environment.NewLine);
-                    }
-                } catch (TaskCanceledException) { } catch (MessageTransportException) { }
-            }).ContinueWith((t) => _shinyRunningTask = null);
+                using (var e = await _workflow.RSession.BeginInteractionAsync()) {
```

Comment:
`MessageTransportException` never comes here, `TaskCanceledException` will just mark task canceled, so there is no need to handle it explicitly

---
In reply to: [71068292](https://github.com/Microsoft/RTVS/pull/2080#discussion_r71068292) [](ancestors = 71068292)